torch/_functorch/_aot_autograd/schemas.py

```diff
@@ -171,46 +171,51 @@ class SubclassCreationMeta:
     flat_tensor_start_idx: int
     # The number of tensors that live in this subclass wrapper
     arg_count: int
+    # meta and inner_keys are produced by the subclass's __tensor_flatten__.
+    # We need to keep them around along with outer_size / outer_stride to plumb them
+    # into __tensor_unflatten__
+    attrs: Dict[str, Union["SubclassCreationMeta", None]]
+    outer_size: List[int]
+    outer_stride: List[int]
+    meta: Any
     # Stores the original subclass itself.
     # This is needed because we need the autograd metadata on the original subclass
     # (this is guaranteed to be a wrapper subclass that holds a fake tensor,
     #  so holding onto this at runtime shouldn't leak memory)
-    original_subclass: torch.Tensor
-    # meta and inner_keys are produced by the subclass's __tensor_flatten__.
-    # We need to keep them around along with outer_size / outer_stride to plumb them
-    # into __tensor_unflatten__.
-    meta: Any
-    inner_keys: List[Any]
-    outer_size: Tuple[int, ...]
-    outer_stride: Tuple[int, ...]
+    original_subclass: Any
+    # If True, then we are constructing metadata at runtime
+    is_runtime: bool
```

Comment:
Hmm we need this flag because we want to construct SubcassCreationMeta at runtime for the backward part to make sure strides/nested-ness behaviour etc didn't change. Only real time this flag is used is for post_init of SubclassCreationMeta where it asserts the original tensor that is used is FakeTensor which is not true for runtime.

Reply:
I think we already do this today? : https://www.internalfb.com/code/fbsource/[70e4cb1526cd6c256299c6eec4c7d40f0367110b]/fbcode/caffe2/torch/_functorch/_aot_autograd/schemas.py?lines=188

Reply:
Ah but in that link, we aren't storing an `is_runtime` field on the metadata - the callee is passing it in directly.

Commented more here: https://github.com/pytorch/pytorch/pull/127431/files#r1651720535